ocs_ci/ocs/osd_operations.py

```diff
@@ -120,6 +120,7 @@ def osd_device_replacement(nodes):
         osd_pod.ocp.wait_for_delete(resource_name=osd_pod_name)
 
     # Run ocs-osd-removal job
+    assert osd_id, "No osd found to remove"
```

Comment:
This `assert` should be performed after the line here https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/osd_operations.py#L69 when we get the osd id.